src/components/screens/SwapScreen/screens/SwapAmountScreen.tsx

```diff
@@ -100,12 +102,49 @@ const SwapAmountScreen: React.FC<SwapAmountScreenProps> = ({
     });
   }, [sourceBalance, account, swapFee]);
 
+  // Check if user has enough XLM for fees when swapping TO XLM
+  const hasXLMForFees = useMemo(() => {
+    if (!destinationBalance || !account) return true;
+
+    // Only check XLM fees when swapping TO XLM
+    if (
+      "token" in destinationBalance &&
+      destinationBalance.token.type === AssetTypeWithCustomToken.NATIVE
+    ) {
+      // Lookup on balanceItems instead of destinationBalance
+      // to avoid using outdated memoized amount
+      const xlmBalance = balanceItems.find(
+        (item) =>
+          "token" in item &&
+          item.token.type === AssetTypeWithCustomToken.NATIVE,
+      );
+
+      if (xlmBalance) {
+        return xlmBalance.total.isGreaterThanOrEqualTo(new BigNumber(swapFee));
+      }
```

Comment:
I'm not sure I understand, I think we need to check that we have enough XLM for fees in all cases, not just when we're swapping _to_ XLM. And importantly, we need to check that we're leaving enough for fees when we're swapping _from_ XLM and we set the "max".

Reply:
I think it's looking good now, as long as it works correctly for the set "max" case, which from reading the code it _looks like_ it's already doing that